src/lib/y2storage/mount_point.rb

```diff
@@ -231,5 +339,34 @@ def passno_must_be_set?
 
       filesystem.type.is?(*TYPES_WITH_PASSNO)
     end
+
+    # DeviceMapper name for the temporary encryption created to calculate the
+    # suitable mount by types
+    TMP_NAME = "dmtemp".freeze
+    private_constant :TMP_NAME
+
+    # Temporary mount point used for the calculation of {#suitable_mount_bys}
+    #
+    # @param encryption [Boolean]
+    # @return [MountPoint]
+    def mount_point_for_suitable(encryption)
+      return self if encryption.nil? || encryption == filesystem.encrypted?
+      return self unless filesystem.is?(:blk_filesystem)
+      # Since it's hard to know what to do in this case...
+      return self if filesystem.multidevice?
+
+      # NOTE: could this short-lived devicegraph be a problem with
+      # the garbage collector?
+      tmp_graph = devicegraph.dup
+      tmp_mount_point = tmp_graph.find_device(sid)
+      tmp_blk_dev = tmp_mount_point.filesystem.blk_devices.first
+      if encryption
+        tmp_blk_dev.create_encryption(TMP_NAME, EncryptionType::PLAIN)
+      else
+        tmp_blk_dev.blk_device.remove_encryption
+      end
+
```

Comment:
Have you tried to call Garbage Collector here? I would say it should fail when accessing `tmp_mount_point.filesystem` outside this method.

Reply:
Correct. That :boom:s